internal/pkg/mapper/mapper.go

```diff
@@ -0,0 +1,228 @@
+package mapper
+
+import (
+	"fmt"
+	"regexp"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	aggregationv1 "github.com/envoyproxy/xds-relay/pkg/api/aggregation/v1"
+)
+
+type Rule = aggregationv1.KeyerConfiguration_Fragment_Rule
+
+type Mapper interface {
+	GetKeys(node core.Node, typeURL string) (string, error)
```

Comment:
I think this param should just be a [v2.DiscoveryRequest](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/discovery.proto#discoveryrequest), since it'll contain the node, type url, resource information, and additional fragment types that we would like to support in the future.

Reply:
i think we should not be working with DiscoveryRequest too deep into our stack since its a transport object. Its better to remove dependency and work with just the resources

Reply:
also ADS depends on typeurl being empty. The typeurl inside DiscoveryRequest might not be enough to make the decision.